src/lib/user-performance-metrics.ts

```diff
@@ -0,0 +1,138 @@
+// Forked and modified from UXM: https://github.com/treosh/uxm
+
+declare let PerformancePaintTiming: any
+
+const perf = typeof window !== "undefined" ? window.performance : null
+
+const sanitizedMetrics = (start, end) =>
+  start > 0 && end > 0 && end - start > 0 ? end - start : null
+
+// user timing helpers
+
+export function mark(markName) {
+  if (perf && perf.mark) {
+    perf.mark(markName)
+  }
+}
+
+export function measure(measureName, startMarkName) {
+  if (perf && perf.measure) {
+    try {
+      perf.measure(measureName, startMarkName)
+    } catch (err) {
+      console.error(err)
+    }
+  }
+}
+
+// default metrics
+
+// export function getEffectiveConnectionType() {
```

Comment:
A few of these metrics (this one and getDeviceMemory) might not really be needed. I may just go ahead and remove them.

Reply:
Hmm, yeah what could we learn from the device’s memory pressure that would be helpful?